src/Field.js

```diff
@@ -1,240 +1,63 @@
-            )
-          : event
-      this.state.state &&
-        this.state.state.change(parse ? parse(value, this.props.name) : value)
-    },
-    onFocus: (event: ?SyntheticFocusEvent<*>) => {
-      this.state.state && this.state.state.focus()
-    }
+  if (typeof component === 'string') {
+    // ignore meta, combine input with any other props
+    return React.createElement(component, { ...field.input, children, ...rest })
   }
+  return renderComponent(
+    { ...field, children, component, ...rest },
+    `Field(${name})`
+  )
+}
 
-  render() {
-    const {
-      allowNull,
-      component,
-      children,
-      format,
-      formatOnBlur,
-      parse,
-      isEqual,
-      name,
-      subscription,
-      validate,
-      validateFields,
-      reactFinalForm,
-      value: _value,
-      ...rest
-    } = this.props
-    let { blur, change, focus, value, name: ignoreName, ...otherState } =
-      this.state.state || {}
-    const meta = {
-      // this is to appease the Flow gods
-      active: otherState.active,
-      data: otherState.data,
-      dirty: otherState.dirty,
-      dirtySinceLastSubmit: otherState.dirtySinceLastSubmit,
-      error: otherState.error,
-      initial: otherState.initial,
-      invalid: otherState.invalid,
-      modified: otherState.modified,
-      pristine: otherState.pristine,
-      submitError: otherState.submitError,
-      submitFailed: otherState.submitFailed,
-      submitSucceeded: otherState.submitSucceeded,
-      submitting: otherState.submitting,
-      touched: otherState.touched,
-      valid: otherState.valid,
-      visited: otherState.visited
-    }
-    if (formatOnBlur) {
-      value = Field.defaultProps.format(value, name)
-    } else if (format) {
-      value = format(value, name)
-    }
-    if (value === null && !allowNull) {
-      value = ''
-    }
-    const input = { name, value, ...this.handlers }
-    if ((rest: Object).type === 'checkbox') {
-      if (_value === undefined) {
-        ;(input: Object).checked = !!value
-      } else {
-        ;(input: Object).checked = !!(
-          Array.isArray(value) && ~value.indexOf(_value)
-        )
-        input.value = _value
-      }
-    } else if ((rest: Object).type === 'radio') {
-      ;(input: Object).checked = value === _value
-      input.value = _value
-    } else if (component === 'select' && (rest: Object).multiple) {
-      input.value = input.value || []
-    }
-
-    if (typeof children === 'function') {
-      return (children: Function)({ input, meta, ...rest })
-    }
-
-    if (typeof component === 'string') {
-      // ignore meta, combine input with any other props
-      return React.createElement(component, { ...input, children, ...rest })
-    }
-    const renderProps: FieldRenderProps = { input, meta } // assign to force Flow check
-    return renderComponent(
-      { ...renderProps, children, component, ...rest },
-      `Field(${name})`
-    )
-  }
+Field.defaultProps = {
```

Comment:
IMHO it's better to just use es6 default params, especially that components with defaultProps might even trigger slower paths inside of react occassionally, so... use the platform 😉 

additionally this is a static property and as I've mentioned this breaks a possibility to tree-shake Field 😢

Reply:
Excellent point. I overlooked this. 👍